test/planwise/component/users_test.clj

```diff
@@ -0,0 +1,90 @@
+(ns planwise.component.users-test
+  (:require [clojure.test :refer :all]
+            [clj-time.core :as time]
+            [clojure.java.jdbc :as jdbc]
+            [com.stuartsierra.component :as component]
+            [planwise.component.users :as users]
+            [planwise.test-system :refer [test-system with-system]]))
+
+(defn execute-sql
+  [system sql]
+  (jdbc/execute! (get-in system [:db :spec]) sql))
+
+(def test-email "user@example.com")
+(def test-full-name "John Doe")
+(def test-user-id 1)
+(def test-scope "resourcemap.instedd.org")
+
+(defn fixture-data
+  []
+  [[:users
+    [{:id test-user-id :email test-email :full_name test-full-name
+      :created_at (time/now)}]]
+   [:tokens
+    [{:user_id test-user-id :scope test-scope :token "EXPIRED-TOKEN-1"
+      :refresh_token "REFRESH-TOKEN-1" :expires (time/ago (time/minutes 5))}
+     {:user_id test-user-id :scope test-scope :token "TOKEN-2"
+      :refresh_token "REFRESH-TOKEN-2" :expires (time/ago (time/minutes 1))}]]])
+
+(defn system
+  []
+  (into
+   (test-system {:fixtures {:data (fixture-data)}})
+   {:users-store (component/using (users/users-store) [:db])}))
+
+(deftest find-user-test
+  (with-system (system)
+    (let [store (:users-store system)
+          user (users/find-user store test-user-id)]
+      (is (some? user))
+      (is (= test-user-id (:id user)))
+      (is (= test-email (:email user)))
+      (is (= test-full-name (:full-name user)))
+      (is (nil? (:last-login user))))))
+
+(deftest find-user-or-create-test
+  (with-system (system)
+    (let [store (:users-store system)]
+      (let [existing (users/find-or-create-user-by-email store test-email)]
+        (is (some? existing))
+        (is (= test-user-id (:id existing)))
+        (is (= test-full-name (:full-name existing))))
+      (let [_ (execute-sql system "ALTER SEQUENCE users_id_seq RESTART WITH 2")
```

Comment:
What if we change `fixtures` to restart all sequences as well as deleting all data on teardown?